winch/codegen/src/abi/mod.rs

```diff
@@ -106,7 +106,7 @@ pub(crate) trait ABI {
     /// Returns the designated scratch register for the given [WasmType].
     fn scratch_for(ty: &WasmType) -> Reg {
         match ty {
-            WasmType::I32 | WasmType::I64 => Self::scratch_reg(),
+            WasmType::I32 | WasmType::I64 | WasmType::Ref(_) => Self::scratch_reg(),
```

Comment:
Only `WasmHeapType::Func` is supported at the moment. Can we update this snippet to reflect that? We'll be able to lift such restriction once externrefs and typed funcs are supported.